lib/fluent/plugin/in_kafka.rb

```diff
@@ -190,7 +190,8 @@ def start
     elsif @username != nil && @password != nil
       @kafka = Kafka.new(seed_brokers: @brokers, client_id: @client_id, logger: logger, ssl_ca_cert: read_ssl_file(@ssl_ca_cert),
                          ssl_client_cert: read_ssl_file(@ssl_client_cert), ssl_client_cert_key: read_ssl_file(@ssl_client_cert_key),
-                         ssl_ca_certs_from_system: @ssl_ca_certs_from_system,sasl_plain_username: @username, sasl_plain_password: @password)
+                         ssl_ca_certs_from_system: @ssl_ca_certs_from_system,sasl_plain_username: @username, sasl_plain_password: @password,
+                         sasl_over_ssl: @sasl_over_ssl)
```

Comment:
I'm not sure but adding the same code at https://github.com/fluent/fluent-plugin-kafka/pull/303/files#diff-16a11d1357395c8e8484ae617f831940R196 is better?

Reply:
I don't think so. In that case neither `scram_mechanism`, `username` or `password` are provided. If I am reading [this](https://github.com/zendesk/ruby-kafka#authentication-using-sasl) correctly, omitting these parameters means that no SASL auth will be attempted so adding `sasl_over_ssl` to that case is not necessary.

Reply:
Make sense.